src/integration/python/knowrob_module.cpp

```diff
@@ -76,6 +78,33 @@ static inline void register_triple_types() {
 	boost::python::class_<TripleList>("TripleList").def(boost::python::vector_indexing_suite<TripleList, true>());
 }
 
+static void InitKnowledgeBaseWrapper(boost::python::list py_argv) {
+	static int argc = boost::python::len(py_argv);
+	static std::vector<char *> argv;
+
+	for (int i = 0; i < argc; ++i) {
+		std::string arg = boost::python::extract<std::string>(py_argv[i]);
```

Comment:
here you copy into a new string `arg` with local allocation within the for loop. so it will be invalid after leaving the current call within the loop